mne/preprocessing/ica.py

```diff
@@ -2084,3 +2101,230 @@ def _band_pass_filter(ica, sources, target, l_freq, h_freq, verbose=None):
+            ax.set_yticks([])
+            ax.set_xticks([])
+            ax.set_frame_on(False)
+        tight_layout(fig=fig)
+        fig.subplots_adjust(top=0.95)
+        fig.canvas.draw()
+        plt.show()
+
+    if threshold == "auto":
+        threshold = np.arange(60, 95, dtype=np.float64) / 100.
+
+    all_maps = [get_ica_map(ica) for ica in icas]
+
+    target = all_maps[template[0]][template[1]]
+
+    if plot is True:
+        ttl = 'Template from subj. {0}'.format(str(template[0]))
+        icas[template[0]].plot_components(picks=template[1],
+                                          ch_type=ch_type,
+                                          title=ttl)
+
+    # first run: use user-selected map
+    if isinstance(threshold, (int, float)):
+        try:
+            nt, mt, s, mx = find_max_corrs(all_maps, target, threshold)
+        except ValueError:
+            logger.info("No component detected using find_outliers."
+                        " Consider using threshold='auto'")
+            return icas
+    elif len(threshold) > 1:
+        paths = [find_max_corrs(all_maps, target, t) for t in threshold]
+        # find iteration with highest avg correlation with target
+        nt, mt, s, mx = paths[np.argmax([path[2] for path in paths])]
+
+    # second run: use output from first run
+    if isinstance(threshold, (int, float)):
+        try:
+            nt, mt, s, mx = find_max_corrs(all_maps, nt, threshold)
+        except ValueError:
+            if threshold > 1:
+                logger.info("No component detected using find_outliers. "
+                            "Consider using threshold='auto'")
+            return icas
+    elif len(threshold) > 1:
+        paths = [find_max_corrs(all_maps, nt, t) for t in threshold]
+        # find iteration with highest avg correlation with target
+        nt, mt, s, mx = paths[np.argmax([path[1] for path in paths])]
+
+    nones = list(range(len(icas)))
+    new_icas, allmaps, indices, subjs = [], [], [], []
+    logger.info("Median correlation with constructed map: %0.3f" % mt)
+    if plot:
+        logger.info("Displaying selected ICs per subject.")
+
+    for ii, (ica, max_corr) in enumerate(zip(icas, mx)):
+        if copy is False:
+            ica = deepcopy(ica)
+        if not hasattr(ica, 'labels_'):
+            ica.labels_ = {}
+        if len(max_corr) > 0:
+            if isinstance(max_corr[0], np.ndarray):
+                max_corr = max_corr[0]
+            ica.labels_[label] = list(set(list(max_corr) +
+                                          ica.labels_.get(label, [])))
+            if plot is True:
+                allmaps.extend(get_ica_map(ica, components=max_corr))
+                subjs.extend([ii] * len(max_corr))
+                indices.extend(max_corr)
+            nones.remove(ii)
+        if copy is False:
+            new_icas.append(ica)
+
+    if plot is True:
+        _plot_corrmap(allmaps, subjs, indices, ch_type, ica, label)
```

Comment:
... basically all figures which are produced inside this hybrid viz-processing code